src/features/unified-share-modal/utils/convertData.js

```diff
@@ -164,3 +164,19 @@ export const convertUserResponse = (userAPIData: User): ContentSharingUserDataTy
         },
     };
 };
+
+/**
+ * Create a shared link permissions object for the API based on a USM permission level.
+ * @param {string} newSharedLinkPermissionLevel
+ */
+export const convertSharedLinkPermissions = (newSharedLinkPermissionLevel: string): $Shape<BoxItemPermission> => {
+    const sharedLinkPermissions = {};
+    Object.keys(USM_TO_API_PERMISSION_LEVEL_MAP).forEach(level => {
```

Comment:
i keep thinking there is a way to optimize this but would need to dig a bit deeper into how exactly.